src/arrayMethodSort.js

```diff
@@ -4,8 +4,53 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  function compareDefault(a, b) {
+    const aString = String(a);
+    const bString = String(b);
+    const areInSameCase =
+      (isUpperCase(aString) && isUpperCase(bString)) ||
+      (!isUpperCase(aString) && !isUpperCase(bString));
+
+    if (areInSameCase) {
+      return aString.localeCompare(b);
+    }
+
+    if (isUpperCase(a)) {
+      return -1;
+    } else {
+      return 1;
+    }
```

Comment:
```suggestion
  if (String(a) > String(b)) {
    return 1;
  }

  if (String(a) < String(b)) {
    return -1;
  }
  
  return 0;
```

Similarly, here we can simplify, we don't need to check whether it is uppercase or not